homeassistant/components/garmin_connect/sensor.py

```diff
@@ -0,0 +1,255 @@
+"""Platform for Garmin Connect integration."""
+from datetime import date, timedelta
+import logging
+
+import garminconnect
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    CONF_EMAIL,
+    CONF_MONITORED_CONDITIONS,
+    CONF_NAME,
+    CONF_PASSWORD,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+GARMIN_CONDITIONS_LIST = {
+    "totalSteps": ["Total Steps", "steps", "mdi:walk"],
+    "dailyStepGoal": ["Daily Step Goal", "steps", "mdi:walk"],
+    "totalKilocalories": ["Total KiloCalories", "kcal", "mdi:food"],
+    "activeKilocalories": ["Active KiloCalories", "kcal", "mdi:food"],
+    "bmrKilocalories": ["BMR KiloCalories", "kcal", "mdi:food"],
+    "consumedKilocalories": ["Consumed KiloCalories", "kcal", "mdi:food"],
+    "burnedKilocalories": ["Burned KiloCalories", "kcal", "mdi:food"],
+    "remainingKilocalories": ["Remaining KiloCalories", "kcal", "mdi:food"],
+    "netRemainingKilocalories": ["Net Remaining KiloCalories", "kcal", "mdi:food"],
+    "netCalorieGoal": ["Net Calorie Goal", "cal", "mdi:food"],
+    "totalDistanceMeters": ["Total Distance Mtr", "mtr", "mdi:walk"],
+    "wellnessStartTimeLocal": ["Wellness Start Time", "", "mdi:clock"],
```

Comment:
Please add support for device classes. For example, this should be `timestamp`. See https://developers.home-assistant.io/docs/en/entity_sensor.html#available-device-classes

Reply:
Done